tools/chezmoi/private/toolchain.bzl

```diff
@@ -0,0 +1,91 @@
+# -*- Starlark -*-
+
+"""Private `chezmoi` toolchain definition."""
+
+def _execute_template_action(
+        ctx,
+        src,
+        out,
+        data,
+        source_dir_files):
+    """
+    Creates a `run_shell` action to execute a chezmoi template.
+    """
+    chezmoi_info = ctx.toolchains["@//tools/chezmoi:toolchain_type"]
+    chezmoi_executable = chezmoi_info.chezmoi_executable
+
+    script_lines = [
+        "#!/bin/bash",
+        "set -euo pipefail",
+
+        # --- Create a self-contained, writable environment for chezmoi ---
+        "export HOME=$(mktemp -d)",
+        "WORK_DIR=$(mktemp -d)",
+
+        # Copy all source files into the working directory.
+        # The -L flag is important to dereference symlinks from the Bazel sandbox.
+        "cp -LR " + "$(dirname " + source_dir_files.to_list()[0].path + ")/." + " $WORK_DIR",
+
+        # Copy the template to be executed into the working directory.
+        "cp -L " + src.path + " $WORK_DIR/" + src.basename,
+
+        # If a data file is provided, copy it to the location where chezmoi expects it.
+    ]
+    if data:
+        script_lines.append("cp -L " + data.path + " $WORK_DIR/.chezmoidata.toml")
```

Comment:
shouldnt we able to accomplish all of this via bazel actions? copy files to an intermediate output dir, then run chezmoi with that as the source directory as cli arg with an output directory as another bazel managed output dir?

i dont see any reaon to need a script here with how chezmoi allows injecting env variables and configuration via the cli